frontend/src/shared/helpers.tsx

```diff
@@ -1038,72 +1077,99 @@ export function prepareSampleMetadataForAgGrid(samples: Sample[]) {
-    const cohortDates = cohorts?.flatMap((c) => {
-      return c.node.hasCohortCompleteCohortCompletes.map((cc) => {
-        return cc.date;
-      });
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      primaryId: sampleMetadata.primaryId,
+      cmoSampleName: sampleMetadata.cmoSampleName,
+      hasStatusStatuses: sampleMetadata.hasStatusStatuses,
+      revisable: s.revisable,
+      ...tempoData,
+    };
+  });
+}
+
+export function prepareCombinedSampleDataForAgGrid(samples: Sample[]) {
+  return samples.map((s) => {
+    const sampleMetadata = s.hasMetadataSampleMetadata[0];
+    const tempoData = extractTempoFromSample(s);
+
+    return {
+      revisable: s.revisable,
+      ...sampleMetadata,
+      ...tempoData,
+    };
+  });
+}
+
+function extractTempoFromSample(s: Sample) {
+  const cohorts = s.cohortsHasCohortSampleConnection?.edges;
+  const cohortDates = cohorts?.flatMap((c) => {
+    return c.node.hasCohortCompleteCohortCompletes.map((cc) => {
+      return cc.date;
     });
-    const deliveryDate = cohortDates?.sort()[0]; // earliest cohort date
+  });
+  let initialPipelineRunDate = cohortDates?.sort()[0]; // earliest cohort date
+  initialPipelineRunDate = formatCohortRelatedDate(initialPipelineRunDate);
 
-    let embargoDate;
-    if (deliveryDate !== undefined) {
-      let embargoDateAsDate = new Date(deliveryDate);
-      embargoDateAsDate.setMonth(embargoDateAsDate.getMonth() + 18);
-      embargoDate = moment(embargoDateAsDate).format("YYYY-MM-DD");
-    }
+  let embargoDate;
+  if (initialPipelineRunDate !== undefined) {
+    let embargoDateAsDate = new Date(initialPipelineRunDate);
+    embargoDateAsDate.setMonth(embargoDateAsDate.getMonth() + 18);
+    embargoDate = moment(embargoDateAsDate).format("YYYY-MM-DD");
+  }
 
-    const tempo = s.hasTempoTempos?.[0];
-    const { billedBy, costCenter, custodianInformation, accessLevel } =
-      tempo ?? {};
+  const tempo = s.hasTempoTempos?.[0];
+  const { billedBy, costCenter, custodianInformation, accessLevel } =
+    tempo ?? {};
 
-    // Without setting null/undefined (falsy) values to false, the Billed column filter will
-    // display "No" 2x when there are both false and null/undefined values in the table.
-    let billed = tempo?.billed;
-    if (!billed) {
-      billed = false;
-    }
+  // Without setting null/undefined (falsy) values to false, the Billed column filter will
+  // display "No" 2x when there are both false and null/undefined values in the table.
+  let billed = tempo?.billed;
+  if (!billed) {
+    billed = false;
+  }
 
-    const bamComplete = tempo?.hasEventBamCompletes?.[0];
-    const { date: bamCompleteDate, status: bamCompleteStatus } =
-      bamComplete ?? {};
+  const bamComplete = tempo?.hasEventBamCompletes?.[0];
+  let { date: bamCompleteDate, status: bamCompleteStatus } = bamComplete ?? {};
```

Comment:
looking at this destructuring here, i'm not sure what it accomplishes.  in other words, where you use these values, why not just use dot notation on the source object.    like, bamComplete.date strikes me as actually better than assigning to bamCompleteDate because then the reader doesn't have to figure out where it's assigned.  sorry if i'm missing something.

Reply:
@alisman We could use individual assignments instead of object destructuring, but ultimately we'll need some unique variable name (`bamCompleteDate` in case) as a field name for the AG Grid column definitions to reference:
```ts
export const CohortSampleDetailsColumns: ColDef[] =
[... 
{
    field: "bamCompleteDate",
    headerName: "Latest BAM Complete Date",
}, 
...]
```

I could update the code like this. Would you prefer this / find it to be more readable?


<table>
<tr>
<th>Before</th>
<th>After</th>
</tr>
<tr>
<td>
  
```ts
  const tempo = s.hasTempoTempos?.[0];
  const { billedBy, costCenter, custodianInformation, accessLevel } =
    tempo ?? {};

  const bamComplete = tempo?.hasEventBamCompletes?.[0];
  let { date: bamCompleteDate, status: bamCompleteStatus } = bamComplete ?? {};
  bamCompleteDate = formatCohortRelatedDate(bamCompleteDate);

  const mafComplete = tempo?.hasEventMafCompletes?.[0];
  let {
    date: mafCompleteDate,
    status: mafCompleteStatus,
    normalPrimaryId: mafCompleteNormalPrimaryId,
  } = mafComplete ?? {};
  mafCompleteDate = formatCohortRelatedDate(mafCompleteDate);

  const qcComplete = tempo?.hasEventQcCompletes?.[0];
  let {
    date: qcCompleteDate,
    result: qcCompleteResult,
    reason: qcCompleteReason,
    status: qcCompleteStatus,
  } = qcComplete ?? {};
  qcCompleteDate = formatCohortRelatedDate(qcCompleteDate);
```
  
</td>
<td>

```ts
  const tempo = s.hasTempoTempos?.[0];
  const billedBy = tempo?.billedBy;
  const costCenter = tempo?.costCenter;
  const custodianInformation = tempo?.custodianInformation;
  const accessLevel = tempo?.accessLevel;

  const bamComplete = tempo?.hasEventBamCompletes?.[0];
  const bamCompleteDate = formatCohortRelatedDate(bamComplete?.date);
  const bamCompleteStatus = bamComplete?.status;

  const mafComplete = tempo?.hasEventMafCompletes?.[0];
  const mafCompleteDate = formatCohortRelatedDate(mafComplete?.date);
  const mafCompleteStatus = mafComplete?.status;
  const mafCompleteNormalPrimaryId = mafComplete?.normalPrimaryId;
  
  const qcComplete = tempo?.hasEventQcCompletes?.[0];
  const qcCompleteDate = formatCohortRelatedDate(qcComplete?.date);
  const qcCompleteResult = qcComplete?.result;
  const qcCompleteReason = qcComplete?.reason;
  const qcCompleteStatus = qcComplete?.status;
```

</td>
</tr>
</table>

Reply:
it is more readable, but a further question is, why are you bothering to assign to local variables here when you are just returning the values below?  i.e. why not just start at line 1155 and return soemthing like 
```
return {
    billed:mafComplete?.whatever
....
}
```